packages/flet/lib/src/utils/overlay_style.dart

```diff
@@ -28,22 +34,20 @@ SystemUiOverlayStyle overlayStyleFromJson(
               ? parseBool(json["enforce_system_navigation_bar_contrast"])
               : null,
       systemNavigationBarIconBrightness: parseBrightness(
-          json["system_navigation_bar_icon_brightness"], brightness),
+          json["system_navigation_bar_icon_brightness"], invertedBrightness),
```

Comment:
I'm still not sure about Flet assuming the icon should be inverse every time.
I tried light green color for status bar and it worked in both light and dark mode, but the icon was "dark" to be visible on light green color.

Let change to  non-inverted brightness? Let's users decide.